### PR TITLE
Openshift

### PR DESCRIPTION
**Pull request by Amirilw**
added openshift option on the values file.
added the needed settings to support openshift.

### CODE REVIEW

File: deploy/charts/kaudit/templates/kaudit.yaml [hunk 1]
```diff
@@ -99,11 +99,18 @@ spec:
       # Pod Security
       serviceAccountName: alcide-k8s-kaudit-{{ .Values.clusterName }}      
       securityContext:
+{{- if eq .Values.k8s.openshift "true" }}
+        # for openshift use be in the ranges: [1000590000, 1000599999]
+        runAsNonRoot: true
+        runAsUser: 1000590000
```
Comment (gadinaor) [inline deploy/charts/kaudit/templates/kaudit.yaml:105]: @Amirilw no need to split for open shift - use 1000590000 instead 10001

File: deploy/charts/kaudit/templates/kaudit.yaml [hunk 2]
```diff
@@ -118,14 +125,19 @@ spec:
...
-        imagePullPolicy: IfNotPresent  
-        securityContext:   
-            allowPrivilegeEscalation: false  
-            capabilities:
-              add:
-                - NET_BIND_SERVICE
-              drop:
-                - ALL                             
+        imagePullPolicy: IfNotPresent
+{{- if eq .Values.k8s.openshift "true" }}
```
Comment (gadinaor) [inline deploy/charts/kaudit/templates/kaudit.yaml:129]: @Amirilw - if the capabilities can be removed for openshift - it is likely that we can drop this in general

This "allowPrivilegeEscalation: false" should stay for both platforms

File: deploy/charts/kaudit/templates/kaudit.yaml [hunk 3]
```diff
@@ -144,23 +156,23 @@ spec:
...
-        imagePullPolicy: {{ .Values.image.pullPolicy }}   
-        securityContext:     
-            allowPrivilegeEscalation: false
-            capabilities:
-                add:
-                - NET_BIND_SERVICE
-                drop:
-                - ALL
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if eq .Values.k8s.openshift "true" }}
```
Comment (gadinaor) [inline deploy/charts/kaudit/templates/kaudit.yaml:164]: @Amirilw same as previous comment

File: deploy/charts/kaudit/values.yaml [hunk 4]
```diff
@@ -83,13 +83,15 @@ aws:
 
 k8s:
   # webhook or auditsink
-  mode: "auditsink" 
+  mode: "auditsink"
+  # if k8s cluster is running via openshift set flag to "true"
+  openshift: "false"
```
Comment (gadinaor) [inline deploy/charts/kaudit/values.yaml:88]: This can be removed once you have the other changes implemented

**Review by gadinaor — CHANGES_REQUESTED** [review]
See comments

File: deploy/charts/kaudit/templates/kaudit.yaml (hunk 1 above)
Comment (Amirilw) [reply deploy/charts/kaudit/templates/kaudit.yaml:105]: this is running with native k8s, a new PR Will be issued

File: deploy/charts/kaudit/templates/kaudit.yaml (hunk 2 above)
Comment (Amirilw) [reply deploy/charts/kaudit/templates/kaudit.yaml:129]: removed

File: deploy/charts/kaudit/templates/kaudit.yaml (hunk 3 above)
Comment (Amirilw) [reply deploy/charts/kaudit/templates/kaudit.yaml:164]: revmoved

File: deploy/charts/kaudit/values.yaml (hunk 4 above)
Comment (Amirilw) [reply deploy/charts/kaudit/values.yaml:88]: removed

**Pull request closed without merging**